### PR TITLE
Tesla : Improve Event stream connection handling and thread pool usage

### PR DESCRIPTION
**Pull request by kgoderis**
Signed-off-by: Karel Goderis <karel.goderis@me.com>

### CODE REVIEW

**Comment by kgoderis** [comment]
@kaikreuzer done.

**Comment by kgoderis** [comment]
Put it on hold - I seem to be using a restricted API - need to fix

**Comment by kgoderis** [comment]
@kaikreuzer Should be good now

File: addons/binding/org.openhab.binding.tesla/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -16,7 +16,7 @@ Import-Package: com.google.common.collect,
  org.eclipse.smarthome.core.thing,
  org.eclipse.smarthome.core.thing.binding,
  org.eclipse.smarthome.core.types,
- org.glassfish.jersey.client.authentication,
+ org.glassfish.jersey.client;version="2.22.1",
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.tesla/META-INF/MANIFEST.MF]: please remove the version constraint

File: addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java [hunk 2]
```diff
@@ -819,23 +829,33 @@ protected void establishEventStream() {
         @Override
         public void run() {
             try {
+
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java]: can you remove this empty line again?

File: addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java [hunk 3]
```diff
@@ -819,23 +829,33 @@ protected void establishEventStream() {
...
                 if (getThing().getStatus() == ThingStatus.ONLINE) {
 
                     if (isAwake()) {
+
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java]: can you remove this empty line again?

**Comment by kgoderis** [comment]
@kaikreuzer minor fixes done

**Pull request merged**